app.py

```diff
@@ -17,92 +46,222 @@ def create_folder(tpost):
+            # handle as a gzip'ed m3u8 file to be downloaded via yt-dlp
+            print('Got gzip\'ed m3u8 file, passing to yt-dlp to save stream:', flush=True)
+            response_m3u8 = requests.get(vpost.url_vid, stream=True)
+            m3u8_path = vpath + '.m3u8'
+            m3u8_file = open(m3u8_path, "wb")
+            m3u8_file.write(response_m3u8.content)
+            m3u8_file.close()
+
+            # generate a file protocol path we can use for yt-dlp. It allows file protocol paths, but only absolute
+            # ones, except if it's on a window drive because the drive letter breaks it, so it has to be absolute to
+            # the current drive, sigh. The actual usuable value is something like `file:///downloads/blah%2Ething.m3u8`
+            full_dir_path = os.path.dirname(pathlib.Path.cwd().joinpath(vpath))
+            file_protocol_path = 'file://' + urllib.parse.quote(str(pathlib.Path(full_dir_path).relative_to(pathlib.Path.cwd().drive)).replace('\\', '/') + '/' + pathlib.Path(m3u8_path).parts[-1])
+
+            cmd = 'yt-dlp.exe -o "' + full_dir_path + '/%(title)s" --retries "infinite" --fragment-retries "infinite" --windows-filenames --no-overwrites --enable-file-urls "' + file_protocol_path + '"'
+            os.system(cmd)
 
-    vpost.url_vid = vidurl.get('1080p', '')
-    vpost.url_vid = vidurl.get('540p', '') if vpost.url_vid == '' else vpost.url_vid
+            print('m3u8 file removed: ' + m3u8_path, flush=True)
+            os.remove(m3u8_path)
+            del response_m3u8
+        else:
+            response_mp4 = requests.get(vpost.url_vid, stream=True)
+            with open(vpath, 'wb') as out_file:
+                shutil.copyfileobj(response_mp4.raw, out_file)
+            del response_mp4
+        del response
 
-    print(f'v: {vpath}')
-    urllib.request.urlretrieve(vpost.url_vid, vpath)
+    except Exception as e:
+        print(e, flush=True)
+        print(logging.traceback.format_exc(), flush=True)
 
 def text_save(tpost):
     tpost.ext = 'txt'
     tpost.prepdata()
 
     folder = create_folder(tpost)
-    tpath = os.path.join(folder, tpost.title)
+    tpath_legacy = os.path.join(folder, cleanFilename(tpost.title_legacy))
+    tpath = os.path.join(folder, cleanFilename(tpost.title))
 
-    print(f't: {tpath}')
+    if not config.overwrite_existing and os.path.exists(tpath_legacy):
+        print(f'v: <<exists skip (legacy filename)>>: {tpath_legacy}', flush=True)
+        append_to_legacy_rename_script_file(folder, tpath_legacy, tpath)
+        return
+    
+    if not config.overwrite_existing and os.path.exists(tpath):
+        print(f'v: <<exists skip>>: {tpath}', flush=True)
+        return
 
     text_file = open(tpath, "w", encoding='utf-8')
     text_file.write(tpost.full_text)
     text_file.close()
 
 def parse_and_get(html_text):
-    soup = BeautifulSoup(html_text, 'html.parser')
+    global skip_next_network_request_delay

```

Comment:
Previously between each entry the script paused, even if the file already existed. This flag is used throughout to skip the next pause when we didn't access the network, making resuming a rip much faster.